libglusterfs/src/dict.c

```diff
@@ -3487,3 +3501,160 @@ dict_has_key_from_array(dict_t *dict, char **strings, gf_boolean_t *result)
     UNLOCK(&dict->lock);
     return 0;
 }
+
+/* Popluate specific dictionary on the basis of passed key array at the
+   time of unserialize buffer
+*/
+int32_t
+dict_unserialize_specific_keys(char *orig_buf, int32_t size, dict_t **fill,
+                               char **specific_key_arr, dict_t **specific_dict)
+{
+    char *buf = orig_buf;
+    int ret = -1;
+    int32_t count = 0;
+    int i = 0;
+    int j = 0;
+
+    data_t *value = NULL;
+    char *key = NULL;
+    int32_t keylen = 0;
+    int32_t vallen = 0;
+    int32_t hostord = 0;
+    xlator_t *this = NULL;
+
+    this = THIS;
+    GF_ASSERT(this);
+
+    if (!buf) {
+        gf_msg_callingfn("dict", GF_LOG_WARNING, EINVAL, LG_MSG_INVALID_ARG,
```

Comment:
any specific reason why are you hardcoding "dict" instead of this->name?

Reply:
We don't call this->name in any other dict_api, i have followed similar code convention.